Define _BSD_SOURCE to make initgroups(3) visible
pipe(2) flags are only used in Linux

Index: util-src/pposix.c
--- util-src/pposix.c.orig
+++ util-src/pposix.c
@@ -31,6 +31,11 @@
 #define _DARWIN_C_SOURCE
 #endif
 #endif
+#if defined(__OpenBSD__)
+#ifndef _BSD_SOURCE
+#define _BSD_SOURCE
+#endif
+#endif
 
 #if ! defined(__FreeBSD__)
 #ifndef _POSIX_C_SOURCE
@@ -654,6 +659,7 @@ static int lc_abort(lua_State *L) {
 	return 0;
 }
 
+#if defined(__linux__)
 const char *pipe_flag_names[] = {
 	"cloexec",
 	"direct",
@@ -664,6 +670,7 @@ const int pipe_flag_values[] = {
 	O_DIRECT,
 	O_NONBLOCK
 };
+#endif
 
 
 static int lc_pipe(lua_State *L) {
